### PR TITLE
🐛Fix subdomain handling in idealmedia widgets

### PR DESCRIPTION
**Pull request by velichkin**
Idealmedia widget wouldn't work if first or second char of data.publisher is not alphanumeric.

### CODE REVIEW

File: ads/idealmedia.js [hunk 1]
```diff
@@ -28,9 +28,11 @@ export function idealmedia(global, data) {
 
   document.body.appendChild(scriptRoot);
 
+  const publisherStr = data.publisher.replace(/[^A-z0-9]/g, '');
```
Comment (choumx) [inline ads/idealmedia.js]: Consider adding a comment explaining this.

**Comment by lannka** [comment]
@velichkin could you address @choumx 's comment?

**Comment by velichkin** [comment]
There is already merged request with totally same changes, but for mgid widgets - https://github.com/ampproject/amphtml/pull/24546/commits/bcdb8c75dedfce5bad09a8db4bf2ad008eced114

About that line of code - it's pretty simple to understand what exactly it is doing, but maybe the puprose of that isn't clear. The idea is that the path to js file containing widget code should be constructed with two first alphanumeric characters of js filename (data.publisher). Should I explain that in a comment?

**Comment by lannka** [comment]
>maybe the puprose of that isn't clear. The idea is that the path to js file containing widget code should be constructed with two first alphanumeric characters of js filename (data.publisher). 

@velichkin yes, it would be more readable if you can put a comment, or even extract a method:

```
function getResourceFilePath(publisher) {
  const publisherStr = publisher.replace(/[^A-z0-9]/g, '');
  return `${publisherStr[0]}/${publisherStr[1])}`;
}
```

When writing this, I noticed the `encodeUriComponent` is not needed as they can only be alphanumeric.

**Comment by lannka** [comment]
now I noticed that `lentainform` & `idealmedia` has mostly identical code. Are they considiered as aliases? If so, we should share the same code.

**Comment by velichkin** [comment]
You are right - lentainform, idealmedia and mgid have mostly identical code. However, they aren't true aliases, so I prefer to handle them separately.

**Comment by velichkin** [comment]
@lannka I've made changes. Is it ok now?

**Pull request merged**